pkg/instrumentation/flags.go

```diff
@@ -16,140 +16,218 @@ package instrumentation
 
 import (
 	"encoding/json"
-	"github.com/intel/cri-resource-manager/pkg/config"
-	"go.opencensus.io/trace"
 	"os"
 	"strconv"
 	"strings"
+	"time"
+
+	"go.opencensus.io/trace"
+
+	"github.com/intel/cri-resource-manager/pkg/config"
+	"github.com/intel/cri-resource-manager/pkg/utils"
 )
 
-// TraceConfig represents a pre-defined instrumentation configuration.
-type TraceConfig float64
+// Sampling defines how often trace samples are taken.
+type Sampling float64
 
 const (
 	// Disabled is the trace configuration for disabling tracing.
-	Disabled TraceConfig = 0.0
+	Disabled Sampling = 0.05
```

Comment:
Hmm, this is bit unclear... how often is there sampling in `Disabled` case (what does 0.05 mean)? Also, if tracing is is disabled, then why a positive number?

Reply:
Symmetric cutoff at both ends of the spectrum. Anything below the lower cutoff becomes 'never/0.0' and anything above the upper cutoff becomes 'always/1.0'. I'm fine with getting rid of both of those. I don't see a point with them in the current code.

Reply:
I'm good with dropping those cut-offs/limits. Up to you, @klihub

Reply:
Dropped both.

Reply:
What were you intending to drop? If the switch-case is preserved,I think the upper end should use `>=`. It isn't very logical that user-defined setting `Sampling=1.0` is treated differently from `Sampling=0.99` or `Sampling=1.01`. Or WDYT?

Reply:
I dropped clamping to 0, or 1 for values <= 0.05 and >= 0.95, which were arbitrary anyway. I checked the code and `trace.ProbabilitySampler()` forces the given probability to 0 or 1 if it is not within the [0,1] range. So now our named values are just recommendations for what the names suggest. But no value is really being treated differently, except the edge cases and only because `trace` defines slightly more efficient dedicated functions for those. Actually `trace.ProbabilitySampler()` internally uses `AlwaysSample()` for values >= 1 but does not use `NeverSample()` for values <= 0... a strange choice IMO. So in principle dropping `case Testing` should have fully identical end result, but I think the code is clearer this way.

Reply:
Thanks for the detailed explanation. So let's drop `case Testing`(?)